site/src/api/queries/groups.ts

```diff
@@ -0,0 +1,101 @@
+import { QueryClient } from "@tanstack/react-query";
+import * as API from "api/api";
+import { checkAuthorization } from "api/api";
+import {
+  CreateGroupRequest,
+  Group,
+  PatchGroupRequest,
+} from "api/typesGenerated";
+
+const GROUPS_QUERY_KEY = ["groups"];
+
+const getGroupQueryKey = (groupId: string) => ["group", groupId];
+
+export const groups = (organizationId: string) => {
+  return {
+    queryKey: GROUPS_QUERY_KEY,
+    queryFn: () => API.getGroups(organizationId),
+  };
+};
+
+export const group = (groupId: string) => {
+  return {
+    queryKey: getGroupQueryKey(groupId),
+    queryFn: () => API.getGroup(groupId),
+  };
+};
+
+export const groupPermissions = (groupId: string) => {
+  return {
+    queryKey: [...getGroupQueryKey(groupId), "permissions"],
+    queryFn: () =>
+      checkAuthorization({
+        checks: {
+          canUpdateGroup: {
+            object: {
+              resource_type: "group",
+              resource_id: groupId,
+            },
+            action: "update",
+          },
+        },
+      }),
+  };
+};
+
+export const createGroup = (queryClient: QueryClient) => {
+  return {
+    mutationFn: ({
+      organizationId,
+      ...request
+    }: CreateGroupRequest & { organizationId: string }) =>
+      API.createGroup(organizationId, request),
+    onSuccess: async () => {
+      await queryClient.invalidateQueries(GROUPS_QUERY_KEY);
```

Comment:
Hm... I just used `await` because eslint was recommending that. Since performance here is not a thing - it is not perceived by the user - I'm good with using whatever the editor - or co-pilot 👀 - suggests.

Reply:
Yeah, that makes sense